laravel-app/tests/Unit/RegisterControllerTest.php

```diff
@@ -49,4 +96,155 @@ public function test_invalid_register_controller_test()
 
         $this->assertTrue($validator->fails());
     }
+
+    public function test_user_email_was_sent_after_registration()
+    {
+        $data = [
+            'first_name' => $this->faker->firstName,
+            'last_name' => $this->faker->lastName,
+            'username' => $this->faker->userName,
+            'email' => $this->faker->unique()->safeEmail,
+            'password' => 'Secret123@Pass',
+            'password_confirmation' => 'Secret123@Pass',
+        ];
+
+        $request = new CreateUserRequest();
+        $validator = validator($data, $request->rules(), $request->messages());
+
+        $this->assertFalse($validator->fails());
+
+        $this->registerUser($data);
+
+        $this->assertEmailSent($data['email']);
+    }
+
+    /** @test */
+    public function test_user_email_not_sent_for_invalid_or_existing_email()
+    {
+        $responseInvalidEmail = $this->post('/register', [
+            'name' => 'Test User',
+            'email' => 'invalid-email',
+            'password' => 'password',
+            'password_confirmation' => 'password',
+        ]);
+
+        $responseInvalidEmail->assertSessionMissing('status');
+
+        $responseExistingEmail = $this->post('/register', [
+            'name' => 'Another User',
+            'email' => 'existing@example.com',
+            'password' => 'password',
+            'password_confirmation' => 'password',
+        ]);
+
+        $responseExistingEmail->assertSessionMissing('status');
+    }
+
+    /** @test */
+    public function test_user_redirected_to_login_after_email_verification_was_sent()
+    {
+        $user = User::create([
+            'username' => 'Test User',
+            'email' => 'test@example.com',
+            'password' => bcrypt('password'),
+            'email_verification_code' => 'sample_verification_code',
+        ]);
+
+        $verificationUrl = route('verify-email', ['verification_code' => $user->email_verification_code]);
+
+        $response = $this->get($verificationUrl);
+
+        $response->assertRedirect('/login');
+    }
```

Comment:
Why is your test not testing your service methods? non-sense right?? you are just testing the default way of creation data in laravel.

Reply:
Apologies sir 🙇 will revise it now 🙇